test/testdrive/kafka-upsert-sources-order-by.td

```diff
@@ -0,0 +1,118 @@
+fish          fish_278        "1970-01-01 00:00:00.278"
+mammal        mammal_278      "1970-01-01 00:00:00.278"
+
+# ingesting again with same timestamp, will remove mammal and update fish
+$ kafka-ingest format=bytes topic=textbytes key-format=bytes key-terminator=: timestamp=278
+mammal:
+fish:new_fish_278
+
+> select key, text, ts from texttext
+key           text            ts
+------------------------------------------------
+fruit         fruit_278       "1970-01-01 00:00:00.278"
+fish          new_fish_278    "1970-01-01 00:00:00.278"
+
+# ingesting with lower timestamp, only bird will be added, fish and fruit will remain unchanged
+$ kafka-ingest format=bytes topic=textbytes key-format=bytes key-terminator=: timestamp=191
+fish:fish_191
+bird:bird_191
+fruit:
+
+> select key, text, ts from texttext
+key           text      ts
+------------------------------------------------
+fruit         fruit_278      "1970-01-01 00:00:00.278"
+fish          new_fish_278   "1970-01-01 00:00:00.278"
+bird          bird_191       "1970-01-01 00:00:00.191"
+
+# Invalid source definitions
+! CREATE SOURCE bad_def FROM KAFKA CONNECTION kafka_conn (TOPIC 'testdrive-textbytes-${testdrive.seed}')
+  KEY FORMAT TEXT VALUE FORMAT TEXT INCLUDE TIMESTAMP
+  ENVELOPE UPSERT (ORDER BY ( KEY )) WITH (SIZE = '1')
+contains:Could not find key in included metadata
+
+! CREATE SOURCE bad_def FROM KAFKA CONNECTION kafka_conn (TOPIC 'testdrive-textbytes-${testdrive.seed}')
+  KEY FORMAT TEXT VALUE FORMAT TEXT INCLUDE TIMESTAMP, PARTITION as p
+  ENVELOPE UPSERT (ORDER BY (p)) WITH (SIZE = '1')
+contains:Only `TIMESTAMP` and `OFFSET` (from `INCLUDE TIMESTAMP` and `INCLUDE OFFSET`) are supported as `ORDER BY` columns
+
+! CREATE SOURCE bad_def FROM KAFKA CONNECTION kafka_conn (TOPIC 'testdrive-textbytes-${testdrive.seed}')
+  KEY FORMAT TEXT VALUE FORMAT TEXT INCLUDE TIMESTAMP as ts, OFFSET
+  ENVELOPE UPSERT (ORDER BY (ts)) WITH (SIZE = '1')
+contains:OFFSET must be explicitly specified as an order by column
+
+# no parentheses around ORDER BY
+! CREATE SOURCE bad_def FROM KAFKA CONNECTION kafka_conn (TOPIC 'testdrive-textbytes-${testdrive.seed}')
+  KEY FORMAT TEXT VALUE FORMAT TEXT INCLUDE TIMESTAMP, PARTITION as p
+  ENVELOPE UPSERT ORDER BY p WITH (SIZE = '1')
+contains:Expected end of statement
+
+# no parentheses around ts in ORDER BY
+! CREATE SOURCE bad_def FROM KAFKA CONNECTION kafka_conn (TOPIC 'testdrive-textbytes-${testdrive.seed}')
+  KEY FORMAT TEXT VALUE FORMAT TEXT INCLUDE TIMESTAMP as ts
+  ENVELOPE UPSERT ( ORDER BY ts ) WITH (SIZE = '1')
+contains:Expected left parenthesis
+
+# DESC ordering not allowed
+! CREATE SOURCE bad_def FROM KAFKA CONNECTION kafka_conn (TOPIC 'testdrive-textbytes-${testdrive.seed}')
+  KEY FORMAT TEXT VALUE FORMAT TEXT INCLUDE TIMESTAMP as ts
+  ENVELOPE UPSERT ( ORDER BY (ts DESC)) WITH (SIZE = '1')
+contains:Expected right parenthesis
+
+# ENVELOPE NONE does not support ORDER BY
```

Comment:
Can you please add such a test for `ENVELOPE DEBEZIUM`, just in case some future refactoring does not suddenly enable this functionality for `DEBEZIUM` as well.

Reply:
Added!